src/core/crash-handler.c

```diff
@@ -100,7 +100,7 @@ _noreturn_ static void crash(int sig, siginfo_t *siginfo, void *context) {
                 } else {
                         siginfo_t status;
 
-                        if (siginfo) {
+                        if (siginfo && siginfo->si_code == SI_USER) {
```

Comment:
This doesn't seem to be complete. The man page indicates that `si_code == SI_QUEUE` after `sigqueue(3)`, and `SI_TKILL` in other circumstances. We want to to print the sender in those circumstances too.